source/common/http/http2/conn_pool.h

```diff
@@ -66,12 +66,29 @@ class ConnPoolImpl : Logger::Loggable<Logger::Id::pool>, public ConnectionPool::
     Upstream::HostDescriptionConstSharedPtr real_host_description_;
     uint64_t total_streams_{};
     Event::TimerPtr connect_timer_;
+    Event::TimerPtr upstream_ready_timer_;
```

Comment:
I somewhat cargo culted this from the http1 implementation, but seems like it was done in order to handle the remote close causing requests being assigned to a connection that's already been closed: https://github.com/envoyproxy/envoy/issues/2715

I think this would apply to http/2 as well? I'll spend some time thinking about it

Reply:
After looking into the issue I don't think we need this for h/2 unless we're already at risk without this PR.  The semantics of how we're handling a connection being closed/reset isn't changing, we're just changing when streams get created on the codec. The issue also talks about how this being due to a h/1.1 issue, though I'm not sure exactly what property of h/1.1 causes this. Maybe the different close semantics vs h/2?

I'll try without using a timer. Hopefully we'll be able to add test coverage that can add confidence around this.

Reply:
I can look at this more tomorrow. For now I would go ahead and remove the timer and we can then think about whether it is necessary.

Reply:
No you don't need this timer when connection is established, this timer is not used in that case in H1 conn pool either.

It is needed in H1 conn pool to schedule another request to existing connection, while the connection might be already closed by upstream so we wait one event cycle. In H2, streams are multiplexed into one connection so the problem doesn't exist. You can call `onUpstreamReady` directly.

Reply:
> There's no similar concern about the primary client's connection being reset immediately after newStream is called?

Can you elaborate a bit more? Connection reset after newStream can happen but the timer cannot save that case either.

Reply:
I think I misunderstood slightly and thought that the timer was meant to help with connections being dropped mid-request. Your explanation makes sense to me now.